lib/service_actor/core.rb

```diff
@@ -57,10 +63,22 @@ def fail!(**arguments)
 
   private
 
+  attr_accessor :service_actor_argument_errors
+
+  def add_argument_errors(add_argument_errors = [])
```

Comment:
Small nitpick - could we rename the argument to sound less actionable? `add_argument_errors` sounds like it could be a proc or lambda.